symengine/expression.h

```diff
@@ -32,16 +32,16 @@ class Expression
     //! Construct Expression from `int`
     Expression(int n) : m_basic(integer(n)) {}
     //! Construct Expression from Basic
-    template <typename T, typename = typename std::enable_if<std::is_constructible<RCP<const Basic>, T &&>::value>::type>
+    template <typename T>
```

Comment:
@certik More generally, you might want to put a check on minimum ICC version if you want to avoid a quagmire of ``ifdef``s. ``noxcept`` was also supported as far as I remember in version 15.